### PR TITLE
 KFSPTS-10829 adding logging to attachment service, and update tests to use mockito

### PR DESCRIPTION
**Pull request by jhulslander**
Please don't merge this yet, still in local validation

### CODE REVIEW

**Comment by jhulslander** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImpl.java [hunk 1]
```diff
@@ -85,6 +85,8 @@ public Attachment createAttachment(PersistableBusinessObject parent, String uplo
             InputStream fileContentsToScan = new ByteArrayInputStream(fileContentsAsByteArray);
 
             if (!ScanResult.Status.PASSED.equals(antiVirusService.scan(fileContentsToScan).getStatus())) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImpl.java]: Should any of the data from the returned ScanResult object be logged as well?

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java [hunk 2]
```diff
@@ -47,28 +50,110 @@
...
+    
+    private InputStream virusInputStream;
+    private InputStream goodInputStream;
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
     @Before
-    public void setUp() throws IOException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    public void setUp() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, IOException{
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Missing space before opening curly brace.

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java [hunk 3]
```diff
@@ -47,28 +50,110 @@
...
+        FileUtils.forceDelete(new File(TEST_PATH).getAbsoluteFile());
+        virusInputStream = null;
+        goodInputStream = null;
+        noteVirus = null;
+        attachmentService = null;
+    }
+    
+    private NoteService buildMockNoteService() {
+        NoteService mockNoteService = Mockito.mock(NoteService.class);
+        Mockito.when(mockNoteService.getNoteByNoteId(Mockito.any())).thenReturn(noteVirus);
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Are there any test cases where you would need to return the "good" note instead?

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java [hunk 4]
```diff
@@ -47,28 +50,110 @@
...
+        return attachment;
+    }
+    
+    private ConfigurationService buildMockConfigurationService() {
+        ConfigurationService mockConfigurationService = Mockito.mock(ConfigurationService.class);
+        Mockito.when(mockConfigurationService.getPropertyValueAsString(Mockito.any())).then(this::getCongigurationPropertyAsString);
+        return mockConfigurationService;
+    }
+    
+    private String getCongigurationPropertyAsString(InvocationOnMock invocation) {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Minor typo in method name.

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java [hunk 5]
```diff
@@ -150,44 +212,45 @@ public void createAttachment() throws Exception {
         Assert.assertEquals("txt", createdAttachment.getAttachmentMimeTypeCode());
         Assert.assertEquals("txt", createdAttachment.getAttachmentTypeCode());
     }
+    
+    @Test
+    public void createAttachmentWithOutVirus() throws Exception {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Is this scenario already being handled by the test case in this class's "createAttachment" test method?

File: src/main/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImpl.java (hunk 1 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImpl.java]: good point, I shall add scan results details

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 2 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: ok

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 3 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: the original code only returned noteVirus for the mock note service.  I think this meets the current testing needs.

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 4 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: ok

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 3 above)
Comment (cah292) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Ok, thanks for clarifying.

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 5 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: The other tests didn't drop into the virus scan's pass result.  This test does that.  I think createAttachment works around the scan code

**Comment by jhulslander** [comment]
Concerning scope, this user story is just for adding logging to CuAttachmentServiceImpl.  If we need more logging on the Cynergy side, it would make sense to have a new user story (if one doesn't already exist) to address them.

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java (hunk 5 above)
Comment (cah292) [reply src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Thanks for checking. Now I see that the test method above this one is using the "dummy" AV service instead, so your new method is indeed testing a new scenario.

**Comment by jhulslander** [comment]
I created KFSPTS-11758 to address improving logging in Cynergy.  If you have details to what changes you think would make sense, please add them.

**Comment by jhulslander** [comment]
this is ready for review again

**Comment by cah292** [comment]
Just to confirm, are these changes ready to be merged once approved? Or are you still performing more local testing as per your first comment?

**Comment by jhulslander** [comment]
This is ready for merge once they pass code review

File: src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java [hunk 6]
```diff
@@ -47,28 +50,110 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        ScanResult result = Mockito.mock(ScanResult.class);
+        
+        if (StringUtils.equals(scannedContents, GOOD_FILE_CONTENTS)) {
+            Mockito.when(result.getStatus()).thenReturn(ScanResult.Status.PASSED);
+        } else {
+            Mockito.when(result.getStatus()).thenReturn(ScanResult.Status.FAILED);
+        }
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java]: Scan results also have an ERROR status. Would it be worth testing that in this test class as well?

**Comment by jhulslander** [comment]
I've added tests for the error response from the virus scan.  This is ready for review again.

**Pull request merged**